tsdb/ooo_head.go

```diff
@@ -26,15 +26,22 @@ import (
 // Samples are stored uncompressed to allow easy sorting.
 // Perhaps we can be more efficient later.
 type OOOChunk struct {
-	samples []sample
+	samples    []sample
+	tombstones *tombstones.MemTombstones
```

Comment:
This is adding 8 bytes per ooo chunk which will increase total memory usage of prometheus instances.

What could be done is having a global ooo tombstones object so we dont need to have one per chunk but then we have an issue when:

```
insertOOO(t0,t1,t2,t3,t4,t5)
delete(t1,t2) // added to global ooo tombstones list
insertOOO(t1,t2)
query(t0,t5) // returns t0,t3,t4,t5 omitting t1,t2 because the global list doesnt know that the last write was after the delete call
```

So I think for now we should add a comment here saying this adds 8 bytes per chunk and a post-optimization can be to have a global list that also tracks ooo chunk references and that will save memory usage.

Reply:
Addressed.